src/Validation/AbstractValidators.php

```diff
@@ -218,7 +220,7 @@ public function create(array $document): ValidatorInterface
     {
         return $this->validatorForResource(
             $this->dataForCreate($document),
-            $this->rules(),
+            $this->rules(null, $document),
```

Comment:
We actually need to pass the document as returned by the `dataForCreate()` method, as that's the data that is passed to the validator.

This is what it needs to be:

```
return $this-validatorForResource(
    $data = $this->dataForCreate($document),
    $this->rules(null, $data),
    $this->messages(),
    $this->attributes()
);
```